src/rpdk/core/fragment/generator.py

```diff
@@ -75,11 +77,43 @@ def validate_fragments(self):
         self.__validate_no_transforms_present(raw_fragments)
         self.__validate_outputs(raw_fragments)
         self.__validate_mappings(raw_fragments)
+        lint_warnings = self.__validate_fragment_through_cfn_lint(raw_fragments)
+        if not lint_warnings:
+            LOG.warning("Module fragment is valid.")
+        else:
+            LOG.warning(
+                "Module fragment is valid, but there are warnings from cfn-lint "
+                "(https://github.com/aws-cloudformation/cfn-python-lint):"
+            )
+        for lint_warning in lint_warnings:
+            print(
+                "\t{} (from rule {})".format(lint_warning.message, lint_warning.rule),
+            )
 
     def __validate_outputs(self, raw_fragments):
         self.__validate_no_exports_present(raw_fragments)
         self.__validate_output_limit(raw_fragments)
 
+    @staticmethod
+    def __validate_fragment_through_cfn_lint(raw_fragment):
+        filename = "temporary_fragment.json"
+
+        with open(filename, "w") as outfile:
+            json.dump(raw_fragment, outfile, indent=4)
+
+        template = cfnlint.decode.cfn_json.load(filename)
```

Comment:
Can we just use the yaml loads() so we don't have to write to a temporary file: https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/decode/cfn_yaml.py#L183?

Reply:
The problem is that `cfnlint.core.run_checks` needs a `filename` to run. I haven't seen an api that validates a template in memory.

Reply:
is filename being used anywhere? 

if it's really necessary recommend: https://docs.python.org/3/library/tempfile.html

Reply:
@jotompki yep, it's used by ` cfnlint.core.run_checks`. I tried using `NamedTemporaryFile`, I don't see the benefit, I have to write the file, then read it back, and then manually remove it?